architecture.md

```diff
@@ -0,0 +1,685 @@
+for example `pc` and `q35` on x86 systems and `virt` on ARM systems. Kata Containers'
+default machine type is `pc`. The default machine type and its [`Machine accelerators`](#Machine-accelerators) can
+be changed by editing the runtime [`configuration`](#Configuration) file.
+
+The follow QEMU features are used in Kata Containers to manage resource constraints, improve
+boot time and reduce memory footprint:
+
+- Machine accelerators.
+- Hot plug devices.
+
+Each feature is documented below.
+
+#### Machine accelerators
+
+Machine accelerators are architecture specific and can used to improve the performance
+and enable specific features of the machine types. The following machine accelerators
+are used in Kata Containers:
+
+- nvdimm: This machine accelerator is x86 specific and only supported by `pc` and
+`q35` machine types. `nvdimm` is used to provide the root filesystem as persistent
+memory device to the Virtual Machine.
+
+Although Kata Containers can run with any recent QEMU release, Kata Containers
+boot time, memory footprint and 9p IO are significantly optimized by using a specific
+QEMU version called [`qemu-lite`](https://github.com/kata-containers/qemu/tree/qemu-lite-2.11.0) and
+custom machine accelerators that are not available in the upstream version of QEMU.
+These custom machine accelerators are described below.
+
+- nofw: this machine accelerator is x86 specific and only supported by `pc` and `q35`
+machine types. `nofw` is used to boot an ELF format kernel by skipping the BIOS/firmware
+in the guest. This custom machine accelerator improves boot time significantly.
+- static-prt: this machine accelerator is x86 specific and only supported by `pc`
+and `q35` machine types. `static-prt` is used to reduce the interpretation burden
+for guest ACPI component.
+
+#### Hot plug devices
+
+The Kata Containers VM starts with a minimum amount of resources, allowing for faster boot time and a reduction in memory footprint.  As the container launch progresses, devices are hotplugged to the VM. For example, when a CPU constraint is specified which includes additional CPUs, they can be hot added.  Kata Containers has support for hot-adding the following devices:
+- Virtio block
+- Virtio SCSI
+- VFIO
+- CPU
+
+### Assets
+
+The hypervisor will launch a virtual machine which includes a minimal guest kernel
+and a guest image.
+
+#### Guest kernel
+
+The guest kernel is passed to the hypervisor and used to boot the virtual
+machine. The default kernel provided in Kata Containers is highly optimized for
+kernel boot time and minimal memory footprint, providing only those services
+required by a container workload. This is based on a very current upstream Linux
+kernel.
+
+#### Guest image
+
+Kata Containers supports both an `initrd` and `rootfs` based minimal guest image.
+
+##### Root filesystem image
+
+The default root filesystem image, sometimes referred to as the "mini O/S", is a
```

Comment:
This section could be reduced to ease maintenance and avoid duplication by referencing the relevant osbuilder docs, namely:

- https://github.com/kata-containers/osbuilder/
- https://github.com/kata-containers/osbuilder/blob/master/image-builder/README.md